store/postgres/migrations/2019-04-19-171709_split_entities/up.sql

```diff
@@ -0,0 +1,106 @@
+-- metadata to track the DB schemas we create for subgraph deployments
+
+-- The name of the DB schema corresponding to a deployment
+create table sgd_schemas(
+  id         serial primary key,
+  subgraph   varchar unique not null,
+  name       varchar not null default 'sgd' || currval('sgd_schemas_id_seq')
+);
+
+insert into sgd_schemas(id, subgraph, name)
+  values(0, 'subgraphs', 'subgraphs');
+
+-- Schema for the subgraph of subgraphs
+-- Note that this schema does not have a entity_history table
+-- because we do not collect history for the subgraph of subgraphs
```

Comment:
It will only freak me out if we collect too much history again ;) As it is right now, we won't collect any history for dynamic data sources, because the triggers are set up to not fire for the `subgraphs` subgraph. I think we should add the `subgraphs.entity_history` table when we address that.

Reply:
"When we address that" — ideally the "when" would be now. If you're saying this already doesn't work on master, then let's create an issue for it, because we're really going to need that for dynamic data sources to be reliable with regards to block reorgs.

Reply:
Yes, it won't work now because the triggers explicitly exclude the `subgraphs` subgraph. That's also what my `TODO` comment was about: I am not sure if `EventSource::None` is a reliable enough indicator whether history should be collected or not. Actually, looking at [this](https://github.com/graphprotocol/graph-node/blob/master/store/postgres/src/store.rs#L904-L913) makes it clear that if we go by `EventSource::None` we will collect history for each block pointer update (or at least the ones that also have entity changes)

Because the right fix seems a little more involved, I would prefer capturing that in a separate issue and addressing it on its own.

Reply:
One more wrinkle: our revert procedure fundamentally assumes we only revert for a single subgraph. With reversions in `subgraphs`, we need to make sure we change that, too.

Reply:
Yep, ok. I've added https://github.com/graphprotocol/graph-node/issues/890 to track this.